ambry-protocol/src/main/java/com/github/ambry/protocol/CompositeSend.java

```diff
@@ -71,8 +71,9 @@ public CompositeSend(List<Send> compositeSendList) {
             if (content instanceof CompositeByteBuf) {
               Iterator<ByteBuf> iterator = ((CompositeByteBuf) content).iterator();
               while (iterator.hasNext()) {
-                ((CompositeByteBuf) compositeSendContent).addComponent(true, iterator.next());
+                ((CompositeByteBuf) compositeSendContent).addComponent(true, iterator.next().retain());
               }
+              content.release();
             } else {
               ((CompositeByteBuf) compositeSendContent).addComponent(true, content);
```

Comment:
why to CompositeByteBufs have to be treated differently here as opposed to just calling `addComponent(true, content)`?

Is there a reason not to nest CompositeByteBufs?

Reply:
I guess my original ideal is not to nest compositebytebuf, I just don't like nesting things. Anyway, it shouldn't be any difference here. I will nest compositebytebuf.